c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1930,38 +1945,22 @@ class DeviceCachingAllocator {
   std::vector<SegmentInfo> snapshot() {
     std::lock_guard<std::recursive_mutex> lock(mutex);
 
-    std::unordered_map<PrivatePool*, MempoolId_t> pool_to_id;
```

Comment:
Once we store the MempoolId_t as a field of the PrivatePool, we don't need to construct this map anymore